AppendStruct() for FixedString does not add Proper Padding

### Issue description
I am using `clickhouse-go/v2` and using `AppendStruct()` to flush rows into a Clickhouse `ReplicatedMergeTree` table. 

When my table has a `FixedString(10)` type defined for a column , the library panics on `AppendStruct()`. 

***Solution*** : I was able to fix this by adding a function to add necessary padding to my strings before passing the pointer to `AppendStruct()`.

The Column Type in Clickhouse is : `Col1 LowCardinality FixedString (7) DEFAULT 'unknown',`

### Example code
```go
type ParquetRow struct {
	Col1       string
}

insert_row := ParquetRow{}
insert_row.Col1 = "Test"

batch, err := conn.PrepareBatch(context.Background(), "INSERT INTO example")
	if err != nil {
		return err
	}

fmt.Println("About to call AppendStruct")

err = batch.AppendStruct(&insert_row)
         if err != nil {
	        log.Fatalf("Failed Appending : ", err)
                return err
	}
```

### Error log
```
About to call AppendStruct
panic: invalid size
goroutine 1 [running]:
[github.com/ClickHouse/ch-go/proto.(*ColFixedStr).Append(...)](http://github.com/ClickHouse/ch-go/proto.(*ColFixedStr).Append(...))
        /Users//go/pkg/mod/[github.com/!click!house/ch-go@v0.51.2/proto/col_fixed_str.go:64](http://github.com/!click!house/ch-go@v0.51.2/proto/col_fixed_str.go:64)
[github.com/ClickHouse/clickhouse-go/v2/lib/column.(*FixedString).AppendRow(0xc0004bd440](http://github.com/ClickHouse/clickhouse-go/v2/lib/column.(*FixedString).AppendRow(0xc0004bd440), {0x1644920?, 0xc000565ab0})
        /Users//go/pkg/mod/[github.com/!click!house/clickhouse-go/v2@v2.6.1/lib/column/fixed_string.go:166](http://github.com/!click!house/clickhouse-go/v2@v2.6.1/lib/column/fixed_string.go:166) +0x37e
[github.com/ClickHouse/clickhouse-go/v2/lib/column.(*LowCardinality).AppendRow(0xc0001270e0](http://github.com/ClickHouse/clickhouse-go/v2/lib/column.(*LowCardinality).AppendRow(0xc0001270e0), {0x1644920, 0xc000565ab0})
        /Users//go/pkg/mod/[github.com/!click!house/clickhouse-go/v2@v2.6.1/lib/column/lowcardinality.go:161](http://github.com/!click!house/clickhouse-go/v2@v2.6.1/lib/column/lowcardinality.go:161) +0x3ad
[github.com/ClickHouse/clickhouse-go/v2/lib/proto.(*Block).Append(0xc00036fd00](http://github.com/ClickHouse/clickhouse-go/v2/lib/proto.(*Block).Append(0xc00036fd00), {0xc000591180?, 0x51, 0x10ce92a?})
        /Users//go/pkg/mod/[github.com/!click!house/clickhouse-go/v2@v2.6.1/lib/proto/block.go:62](http://github.com/!click!house/clickhouse-go/v2@v2.6.1/lib/proto/block.go:62) +0x1a7
[github.com/ClickHouse/clickhouse-go/v2.(*batch).Append(0xc000622cd0](http://github.com/ClickHouse/clickhouse-go/v2.(*batch).Append(0xc000622cd0), {0xc000591180?, 0xc?, 0xc000590c00?})
        /Users//go/pkg/mod/[github.com/!click!house/clickhouse-go/v2@v2.6.1/conn_batch.go:122](http://github.com/!click!house/clickhouse-go/v2@v2.6.1/conn_batch.go:122) +0x48
[github.com/ClickHouse/clickhouse-go/v2.(*batch).AppendStruct(0xc000622cd0](http://github.com/ClickHouse/clickhouse-go/v2.(*batch).AppendStruct(0xc000622cd0), {0x1635100?, 0xc0005faa00?})
        /Users//go/pkg/mod/[github.com/!click!house/clickhouse-go/v2@v2.6.1/conn_batch.go:138](http://github.com/!click!house/clickhouse-go/v2@v2.6.1/conn_batch.go:138) +0x95
main.main()
       insert_clickhouse/main.go:263 +0x1188
exit status 2
```

--- a/lib/binary/string_unsafe.go
+++ b/lib/binary/string_unsafe.go
@@ -20,6 +20,14 @@
 
 package binary
 
-func Str2Bytes(str string) []byte {
-	return unsafeStr2Bytes(str)
+func Str2Bytes(str string, expectedLen int) []byte {
+	b := unsafeStr2Bytes(str)
+
+	if len(str) < expectedLen {
+		extended := make([]byte, expectedLen)
+		copy(extended, b)
+		return extended
+	}
+
+	return b
 }

--- a/lib/column/fixed_string.go
+++ b/lib/column/fixed_string.go
@@ -97,7 +97,7 @@ func (col *FixedString) Append(v interface{}) (nulls []uint8, err error) {
 			if v == "" {
 				col.col.Append(make([]byte, col.col.Size))
 			} else {
-				col.col.Append(binary.Str2Bytes(v))
+				col.col.Append(binary.Str2Bytes(v, col.col.Size))
 			}
 		}
 	case []*string:
@@ -113,7 +113,7 @@ func (col *FixedString) Append(v interface{}) (nulls []uint8, err error) {
 				if *v == "" {
 					col.col.Append(make([]byte, col.col.Size))
 				} else {
-					col.col.Append(binary.Str2Bytes(*v))
+					col.col.Append(binary.Str2Bytes(*v, col.col.Size))
 				}
 			}
 		}
@@ -139,12 +139,12 @@ func (col *FixedString) AppendRow(v interface{}) (err error) {
 	switch v := v.(type) {
 	case string:
 		if v != "" {
-			data = binary.Str2Bytes(v)
+			data = binary.Str2Bytes(v, col.col.Size)
 		}
 	case *string:
 		if v != nil {
 			if *v != "" {
-				data = binary.Str2Bytes(*v)
+				data = binary.Str2Bytes(*v, col.col.Size)
 			}
 		}
 	case nil:

--- a/lib/column/string.go
+++ b/lib/column/string.go
@@ -71,7 +71,7 @@ func (col *String) ScanRow(dest interface{}, row int) error {
 	case *sql.NullString:
 		return d.Scan(val)
 	case encoding.BinaryUnmarshaler:
-		return d.UnmarshalBinary(binary.Str2Bytes(val))
+		return d.UnmarshalBinary(binary.Str2Bytes(val, len(val)))
 	default:
 		if scan, ok := dest.(sql.Scanner); ok {
 			return scan.Scan(val)

--- a/tests/fixed_string_test.go
+++ b/tests/fixed_string_test.go
@@ -53,6 +53,7 @@ func TestFixedString(t *testing.T) {
 				, Col3 Nullable(FixedString(10))
 				, Col4 Array(FixedString(10))
 				, Col5 Array(Nullable(FixedString(10)))
+			    , Col6 FixedString(12)
 			) Engine MergeTree() ORDER BY tuple()
 		`
 	defer func() {
@@ -67,10 +68,11 @@ func TestFixedString(t *testing.T) {
 		col3Data = &col1Data
 		col4Data = []string{"ClickHouse", "ClickHouse", "ClickHouse"}
 		col5Data = []*string{&col1Data, nil, &col1Data}
+		col6Data = "clickhouse"
 	)
 	_, err = rand.Read(col2Data.data[:])
 	require.NoError(t, err)
-	require.NoError(t, batch.Append(col1Data, col2Data, col3Data, col4Data, col5Data))
+	require.NoError(t, batch.Append(col1Data, col2Data, col3Data, col4Data, col5Data, col6Data))
 	require.NoError(t, batch.Send())
 	var (
 		col1 string
@@ -78,13 +80,15 @@ func TestFixedString(t *testing.T) {
 		col3 *string
 		col4 []string
 		col5 []*string
+		col6 string
 	)
-	require.NoError(t, conn.QueryRow(ctx, "SELECT * FROM test_fixed_string").Scan(&col1, &col2, &col3, &col4, &col5))
+	require.NoError(t, conn.QueryRow(ctx, "SELECT * FROM test_fixed_string").Scan(&col1, &col2, &col3, &col4, &col5, &col6))
 	assert.Equal(t, col1Data, col1)
 	assert.Equal(t, col2Data.data, col2.data)
 	assert.Equal(t, col3Data, col3)
 	assert.Equal(t, col4Data, col4)
 	assert.Equal(t, col5Data, col5)
+	assert.Equal(t, col6Data+string([]byte{0, 0}), col6)
 	rows, err := conn.Query(ctx, "SELECT CAST('RU' AS FixedString(2)) FROM system.numbers_mt LIMIT 10")
 	require.NoError(t, err)
 	var count int

--- a/tests/issues/904_test.go
+++ b/tests/issues/904_test.go
@@ -1,0 +1,62 @@
+// Licensed to ClickHouse, Inc. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. ClickHouse, Inc. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package issues
+
+import (
+	"context"
+	"fmt"
+	"github.com/ClickHouse/clickhouse-go/v2"
+	clickhouse_tests "github.com/ClickHouse/clickhouse-go/v2/tests"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func Test904(t *testing.T) {
+	var (
+		conn, err = clickhouse_tests.GetConnection("issues", clickhouse.Settings{
+			"max_execution_time": 60,
+			"flatten_nested":     0,
+		}, nil, &clickhouse.Compression{
+			Method: clickhouse.CompressionLZ4,
+		})
+	)
+
+	ctx := context.Background()
+	require.NoError(t, err)
+	env, err := clickhouse_tests.GetTestEnvironment(testSet)
+	require.NoError(t, err)
+
+	ddl := fmt.Sprintf("CREATE TABLE `%s`.`test_904` (Col1 FixedString(6)) Engine MergeTree() ORDER BY tuple()", env.Database)
+	defer func() {
+		conn.Exec(ctx, fmt.Sprintf("DROP TABLE IF EXISTS `%s`.`test_904`", env.Database))
+	}()
+	require.NoError(t, conn.Exec(ctx, ddl))
+
+	batch, err := conn.PrepareBatch(ctx, fmt.Sprintf("INSERT INTO `%s`.`test_904` (Col1)", env.Database))
+	require.NoError(t, err)
+	require.NoError(t, batch.Append("foo"))
+	require.NoError(t, batch.Send())
+
+	var col1 string
+	row := conn.QueryRow(ctx, fmt.Sprintf("SELECT Col1 FROM `%s`.`test_904`", env.Database))
+	require.NoError(t, row.Err())
+	require.NoError(t, row.Scan(&col1))
+
+	assert.Equal(t, "foo"+string([]byte{0, 0, 0}), col1)
+}

